lib/spack/spack/spec.py

```diff
@@ -1267,6 +1267,7 @@ def cshort_spec(self):
 
     @property
     def prefix(self):
+        assert self._concrete
```

Comment:
change to a conditional and a `SpecError` to be consistent with other methods in `spec.py`